platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceImpl.java

```diff
@@ -193,17 +196,14 @@ public ApplicationStatus getApplicationStatus(Application application) {
         isMainFeatureUninstalled = !featuresService.isInstalled(mainFeature);
         requiredFeatures.add(mainFeature);
       } else {
-
         Set<Feature> features = application.getFeatures();
         if (features.size() == 1) {
           requiredFeatures.addAll(getAllDependencyFeatures(features.iterator().next()));
         } else {
-          for (Feature curFeature : features) {
-            if (StringUtils.equalsIgnoreCase(
-                Feature.DEFAULT_INSTALL_MODE, curFeature.getInstall())) {
-              requiredFeatures.addAll(getAllDependencyFeatures(curFeature));
-            }
-          }
+          features
+              .stream()
+              .filter(f -> equalsIgnoreCase(Feature.DEFAULT_INSTALL_MODE, f.getInstall()))
+              .forEach(f -> requiredFeatures.addAll(getAllDependencyFeatures(f)));
```

Comment:
✏️  Could be replace with ```.map(this::getAllDependencyFeatures).forEach(requiredfeatures::addAll)```

Reply:
Even better would be to collect that as a set and assign it to `requiredFeatures` instead of adding to it.